identifier/src/containers/Login/Login.jsx

```diff
@@ -81,9 +81,16 @@ function Login(props) {
       }
 
       history.replace(`/chooseaccount${history.location.search}${history.location.hash}`);
-      return;
-    }
-  }, [ /* no dependencies */ ]); // eslint-disable-line react-hooks/exhaustive-deps
+        return;
+      }
+    if (
+      query &&
+      query.login_hint &&
+      (validator.isEmail(query.login_hint) || validator.isEmail(`${query.login_hint}@example.com`))) {
```

Comment:
Please simplify this if, split it into two ifs (first checking for `query.login_hint` making use of typescript ? stuff to avoid writing an additional check for query) (i am not sure why you check on query in the first place, can this even be undefined or null, typescript should tell). It is also not checked in the useEffect above.

Reply:
Yes this can be undefined based on user's inaddition of login_hint. That's why the check has been added. And seperated the if statement into two lines.